qemu/tests/fio_windows.py

```diff
@@ -33,31 +33,45 @@ def run(test, params, env):
 
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
-    session = vm.wait_for_login(timeout=timeout)
+    session_data = vm.wait_for_login(timeout=timeout)
+    session_sys = vm.wait_for_login(timeout=timeout)
+
+    if params.get("format_disk", "no") == "yes":
+        error_context.context("Format disk", logging.info)
+        utils_misc.format_windows_disk(session_data, params["disk_index"],
+                                       mountpoint=params["disk_letter"])
     try:
-        installed = session.cmd_status(check_installed_cmd) == 0
+        installed = session_data.cmd_status(check_installed_cmd) == 0
         if not installed:
-            dst = r"%s:\\" % utils_misc.get_winutils_vol(session)
+            dst = r"%s:\\" % utils_misc.get_winutils_vol(session_data)
 
             error_context.context("Install fio in guest", logging.info)
             install_cmd = params["install_cmd"]
             install_cmd = re.sub(r"DRIVE:\\+", dst, install_cmd)
-            session.cmd(install_cmd, timeout=180)
+            session_data.cmd(install_cmd, timeout=180)
             time.sleep(30)
             config_cmd = params.get("config_cmd")
             if config_cmd:
-                session.cmd(config_cmd)
+                session_data.cmd(config_cmd)
 
         error_context.context("Start fio in guest.", logging.info)
-        s, o = session.cmd_status_output(fio_cmd, timeout=(cmd_timeout+60))
-        if s:
-            raise exceptions.TestError("Failed to run fio, output: %s" % o)
+        fio_thread_sys = utils_misc.InterruptedThread(session_sys.cmd_status_output,
+                                                      (fio_cmd_sys, (cmd_timeout+120)))
+        fio_thread_sys.start()
+        status_data, output_data = session_data.cmd_status_output(fio_cmd_data,
+                                                                  timeout=(cmd_timeout+480))
+        status_sys, output_sys = fio_thread_sys.join()
+        if status_data or status_sys:
+            raise test.error("Failed to run fio, output: %s\n%s"
```

Comment:
you don't need raise here, just test.error will do

Reply:
ok, will change it, thanks~